### PR TITLE
103965 - IPO Show more data in history. Using new extension class to h…

### PR DESCRIPTION
**Pull request by trondlindanger**
…andle user info in combination with event type.
[AB#103965](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/103965)

### CODE REVIEW

**Comment by bratli** [comment]
Add link to PBI by adding AB#XXXXX in the description, where XXXXX is the PBI number

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/HistoryAggregate/EventTypeExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.PersonAggregate;
+
+namespace Equinor.ProCoSys.IPO.Domain.AggregateModels.HistoryAggregate
+{
+    public static class EventTypeExtensions
+    {
+        public static string GetDescription(this EventType eventType, Participant participant)
+        {
+            string user = $"{participant.FirstName} {participant.LastName}";
+            string functionRole = participant.FunctionalRoleCode;
```
Comment (bratli) [inline src/Equinor.ProCoSys.IPO.Domain/AggregateModels/HistoryAggregate/EventTypeExtensions.cs:13]: nit: it is called functionalRole

File: src/Equinor.ProCoSys.IPO.Domain/Events/PreSave/IpoCompletedEvent.cs [hunk 2]
```diff
@@ -1,16 +1,20 @@
 ﻿using System;
 using Equinor.ProCoSys.Common;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
 
 namespace Equinor.ProCoSys.IPO.Domain.Events.PreSave
 {
     public class IpoCompletedEvent : IDomainEvent
     {
-        public IpoCompletedEvent(string plant, Guid sourceGuid)
+        public IpoCompletedEvent(string plant, Guid sourceGuid, Participant participant = null)
```
Comment (bratli) [inline src/Equinor.ProCoSys.IPO.Domain/Events/PreSave/IpoCompletedEvent.cs:9]: Is there a scenario where participant ends up being null? Will we then get a NPE in EventTypeExtensions when having participant.firstName?

File: src/tests/Equinor.ProCoSys.IPO.Command.Tests/EventHandlers/HistoryEvents/AttendedStatusUpdatedEventHandlerTest.cs [hunk 3]
```diff
@@ -38,6 +39,17 @@ public void Handle_ShouldUpdateAttendedStatusUpdatedHistoryRecord()
             // Act
             var sourceGuid = Guid.NewGuid();
             var plant = "TestPlant";
+            var participant = new Participant("TestPlant", 
+                                    Domain.AggregateModels.InvitationAggregate.Organization.ConstructionCompany,
```
Comment (bratli) [inline src/tests/Equinor.ProCoSys.IPO.Command.Tests/EventHandlers/HistoryEvents/AttendedStatusUpdatedEventHandlerTest.cs:43]: can we have this as a using instead?

Comment (bratli) [reply src/tests/Equinor.ProCoSys.IPO.Command.Tests/EventHandlers/HistoryEvents/AttendedStatusUpdatedEventHandlerTest.cs:43]: same for all files

File: src/tests/Equinor.ProCoSys.IPO.Command.Tests/EventHandlers/HistoryEvents/AttendedStatusUpdatedEventHandlerTest.cs [hunk 4]
```diff
@@ -36,29 +36,29 @@ public void Handle_ShouldUpdateAttendedStatusUpdatedHistoryRecord()
...
 
-            // Assert
-            Assert.IsNotNull(_historyAdded);
-            Assert.AreEqual(plant, _historyAdded.Plant);
-            Assert.AreEqual(sourceGuid, _historyAdded.SourceGuid);
-            Assert.IsNotNull(_historyAdded.Description);
-            Assert.AreEqual(EventType.AttendedStatusUpdated, _historyAdded.EventType);
-            Assert.AreEqual("IPO", _historyAdded.ObjectType);
+            // Act
+            using (_dut.Handle(new AttendedStatusUpdatedEvent(plant, sourceGuid), default)) {
```
Comment (bratli) [inline src/tests/Equinor.ProCoSys.IPO.Command.Tests/EventHandlers/HistoryEvents/AttendedStatusUpdatedEventHandlerTest.cs:53]: why do we need this "using"?

File: src/Equinor.ProCoSys.IPO.Domain/Events/PreSave/IpoCompletedEvent.cs (hunk 2 above)
Comment (trondlindanger) [reply src/Equinor.ProCoSys.IPO.Domain/Events/PreSave/IpoCompletedEvent.cs]: Originally necessary, but not in the final solution. Removed.

**Review by bratli — APPROVED** [review]

**Pull request merged**